src/_arraykit.c

```diff
@@ -3505,6 +3517,52 @@ array_deepcopy(PyObject *m, PyObject *args, PyObject *kwargs)
     return AK_ArrayDeepCopy(m, (PyArrayObject*)array, memo);
 }
 
+// JTODO: here
+// Reshape if necessary a row that might be 2D or 1D is returned as a 1D array.
+static PyObject *
+array2d_to_array1d(PyObject *Py_UNUSED(m), PyObject *a)
+{
+    AK_CHECK_NUMPY_ARRAY_2D(a); // JTODO: should we allow 1D arrays?
+    PyArrayObject *input_array = (PyArrayObject *)a;
+    
+    // get array dimensions
+    npy_intp num_rows = PyArray_DIM(input_array, 0);
+    npy_intp num_cols = PyArray_DIM(input_array, 1);
+
+    // create output array
+    npy_intp dims[1] = {num_rows};
+    PyObject* output_array = PyArray_SimpleNew(1, dims, NPY_OBJECT);
+    if (!output_array) {
+        return NULL; // JTODO: check if this is the correct error handling
+    }
+
+    for (npy_intp i = 0; i < num_rows; ++i) {
+        PyObject* tuple = PyTuple_New(num_cols);
+        if (!tuple) {
+            Py_DECREF(output_array);
+            return NULL;
+        }
+
+        for (npy_intp j = 0; j < num_cols; ++j) {
+            PyObject* item = PyArray_GETITEM(input_array, PyArray_GETPTR2(input_array, i, j));
+            if (!item) {
+                Py_DECREF(tuple);
+                Py_DECREF(output_array);
+                return NULL;
+            }
+            PyTuple_SET_ITEM(tuple, j, item);
+        }
+
+        PyArray_SETITEM((PyArrayObject*)output_array, PyArray_GETPTR1((PyArrayObject*)output_array, i), tuple);
```

Comment:
Rather than use `PyArray_SETITEM`, we can take advantage of the fact that we have just created this empty object array (with `PyArray_SimpleNew`) and we know that it is contiguous in memory. This lets use the fastest path of iterative assignment to an array: pointer arithmetic and direct assignment to the underlying buffer.

Your outer loop will look something like this:
```c
PyObject** output_data = (PyObject**)PyArray_DATA(output_array);
PyObject** p = output_data;
PyObject** p_end = p + num_rows;

while (p < p_end) {
    // create your tuple
    *p++ = tuple;
}